File: src/main/java/ordilov/randomplay/like/domain/LikedTrack.java
package ordilov.randomplay.like.domain;

import static javax.persistence.GenerationType.IDENTITY;
import static lombok.AccessLevel.PROTECTED;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import lombok.Getter;
import lombok.NoArgsConstructor;
import ordilov.randomplay.member.domain.Member;
import ordilov.randomplay.playlist.domain.Playlist;
import ordilov.randomplay.track.domain.Track;

@Entity
@Getter
@NoArgsConstructor(access = PROTECTED)
public class LikedTrack {

  @Id
  @GeneratedValue(strategy = IDENTITY)
  private Long id;

  @ManyToOne
  @JoinColumn(name = "member_id")
  private Member member;

  @ManyToOne
  @JoinColumn(name = "track_id")
  private Track track;

  public LikedTrack(Member member, Track track) {
    this.member = member;
    this.track = track;
    this.track.addLikeCount();
  }
}

File: src/main/java/ordilov/lissn/like/domain/LikedTrack.kt
package ordilov.lissn.like.domain

import ordilov.lissn.member.domain.Member
import ordilov.lissn.track.domain.Track
import javax.persistence.*

@Entity
class LikedTrack(
    @ManyToOne
    @JoinColumn(name = "member_id")
    val member: Member,

    @ManyToOne
    @JoinColumn(name = "track_id")
    val track: Track,

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
)